src/main/java/it/pagopa/pn/deliverypush/service/impl/WebhookServiceImpl.java

```diff
@@ -52,7 +53,24 @@ private Mono<StreamEntity> filterEntity(String xPagopaPnApiVersion, String xPago
                 || (
                     mode == StreamEntityAccessMode.WRITE ?  WebhookUtils.checkGroups(streamEntity.getGroups(), xPagopaPnCxGroups) : WebhookUtils.checkGroups(xPagopaPnCxGroups, streamEntity.getGroups())
                 )
-            ).filter(streamEntity -> ignoreVersion
+            )
+            .switchIfEmpty(Mono.error(new PnWebhookForbiddenException("Pa " + xPagopaPnCxId + " groups (" + join(xPagopaPnCxGroups)+ ") is not allowed to see this streamId " + streamId)))
+            .filter(streamEntity ->{
+                if (apiV10.equals(entityVersion(streamEntity)) && ignoreVersion){
+                    return true;
+                }
+
+                //Se non sono master non posso agire in scrittura su stream senza gruppi: solo per v23
+                if ((!apiV10.equals(apiVersion(xPagopaPnApiVersion)) && !apiV10.equals(entityVersion(streamEntity)))
```

Comment:
Sonar qui rileva: "Return of boolean expressions should not be wrapped into an "if-then-else" statement". Possiamo wrappare queste condizioni in un metodo privato (con nome parlante) e ritornare direttamente quest'ultimo?